app/src/main/java/swati4star/createpdf/Home.java

```diff
@@ -144,6 +158,23 @@ void startAddingImages() {
         }
     }
 
+    void cropImages() {
+        int imageCount = 1;
+        if (tempUris.size() == 0) {
+            Toast.makeText(activity, R.string.toast_no_images, Toast.LENGTH_SHORT).show();
+            return;
+        }
+        for (String imageUri : tempUris) {
+            CropImage.activity(Uri.fromFile(new File(imageUri)))
```

Comment:
This might slow down or hang the phone, in case I select 20 images.

Reply:
Do you have a better idea for handling this? I could try to fix this in a way that the next activity only gets started if the previous one has finished.

Reply:
Fixed this issue: There will always only be one CropImage activity running as the next activity gets called only after the getting result of the previous one